testutil/integration/suites/base_test.go

```diff
@@ -0,0 +1,217 @@
+//go:build integration
+
+package suites
+
+import (
+	"testing"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/pokt-network/poktroll/app/volatile"
+	"github.com/pokt-network/poktroll/testutil/events"
+	"github.com/pokt-network/poktroll/testutil/integration"
+	"github.com/pokt-network/poktroll/testutil/sample"
+	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
+)
+
+type BaseIntegrationSuiteTestSuite struct {
```

Comment:
Confirming that this (and othesr) need to be publically exposed?

Can you add a godoc for all of them please. It'll help when we try to get others to adopt this.

Reply:
```go
// baseIntegrationSuiteTestSuite is a test suite which embeds BaseIntegrationSuite.
// **in order to test it**. It is intended to be embedded in other test suites.
type baseIntegrationSuiteTestSuite struct {
	BaseIntegrationSuite
}
```

Reply:
Are you saying you would like to see godoc comments on the test methods?

Reply:
1. The struct yets.
2. The methods. Yes?

My rationale is that you built a lot of helpers but I'm afraid I won't know how/when to use them and am beginning to think that [1] will be the way to go.

[1] https://pkg.go.dev/github.com/pokt-network/poktroll

Reply:
I agree that the godocs are the way to go but I think we're talking past each other a bit here.

This is "test code", as in code that testing the thing that we would actually want to have godocs for. **Test code does is omitted from godoc output, so any godoc-style comments added here are only for the benefit of someone looking at this code** (with the exception of [Testable Examples](https://go.dev/blog/examples)).

I added the comment above to the struct and unexported it to avoid confusion. The object under test (`BaseIntegrationSuite`) has godoc comments for its struct and methods. Is that sufficient?

Reply:
UPDATE: I'm realizing there was a typo in the comment, I most unfortunately omitted the word "NOT". :sweat_smile:. Corrected.